sdks/python/apache_beam/runners/worker/log_handler.py

```diff
@@ -108,5 +113,16 @@ def _write_log_entries(self):
 
   def _read_log_control_messages(self, log_control_iterator):
     # TODO(vikasrk): Handle control messages.
-    for _ in log_control_iterator:
-      pass
+    while True:
+      try:
+        for _ in log_control_iterator:
+          pass
+        # iterator is closed
+        return
+      except Exception as ex:
+        # reset the stream
+        print >> sys.stderr, "Logging client failed: {}... resetting".format(ex)
+        self.acquire()
```

Comment:
is this to prevent emit from running while this is happening?

Reply:
Yeah, that was the idea. However digging a bit deeper I think this is undesirable, as it will block application logic during the reconnection process. Given that we have a queue in between the emit call and the grpc stream, it shouldn't be necessary to lock.

Reply:
However, currently the queue is unbounded, so if we remove this we'd need to bound the queue and either block when it's full or drop messages. Given that logging is a secondary concern, I think it may make more sense to drop the messages. Thoughts?

Reply:
It sounds reasonable to me to put an upper bound on logs that might be buffered when disconnected. 
- The upper bound of 1000 messages is arbitrary (any number will be arbitrary though). If we take your estimate of 1K/log message, we can probably increase that a bit.
- It would help to have a log message after reconnecting saying that some message might have been dropped in between. Most people reading a set of logs will assume its completeness.

Reply:
Yeah, happy to change the buffer size according to what level of memory usage people find reasonable. 10,000?

I like the idea of logging a message on reconnection. Ideally we'd also have a metric for tracking dropped logs, but my understanding is that there isn't currently metric support in the python SDK.